be/src/runtime/fragment_mgr.h

```diff
@@ -152,6 +152,11 @@ class FragmentMgr : public RestMonitorIface {
         return _query_ctx_map.size();
     }
 
+    std::unordered_map<TUniqueId, std::shared_ptr<QueryContext>>& get_query_ctx_map() {
```

Comment:
never return this map, because you add lock to return a reference,the map maybe modified concurrently.

I think you could add a method named colloect_query_statistics in FragmentMgr to avoid this.